### PR TITLE
[DOC] Specify upgrade order for tiers

### PR DESCRIPTION
**Pull request by pxsalehi**
Outdated info in comparison to https://www.elastic.co/guide/en/elastic-stack/8.7/upgrading-elasticsearch.html#upgrading-elasticsearch

### CODE REVIEW

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 1]
```diff
@@ -7,19 +7,22 @@ a time so upgrading does not interrupt service. Running multiple versions of
...
 of these nodes with `GET /_nodes/_all,master:false` or by finding all the nodes
 configured with `node.master: false`.
 
-. Master-eligible nodes, which are the remaining nodes. You can retrieve a list
-of these nodes with `GET /_nodes/master:true`.
+. Upgrade nodes tier-by-tier, starting with the frozen tier.
+Complete the upgrade for all nodes in each data tier before moving to the next.
+Upgrade the frozen tier, then the cold tier, then the warm tier, and upgrade
+the hot tier last. This ensures {ilm-init} can continue to move data through
+the tiers during the upgrade.
```
Comment (DaveCTurner) [inline docs/reference/upgrade/rolling_upgrade.asciidoc:20]: You (possibly deliberately) omitted this sentence, but I think it's helpful:

```suggestion
the tiers during the upgrade. You can get the list of nodes in a specific tier
with a `GET /_nodes` request, for example: `GET /_nodes/data_frozen:true/_none`.
```

**Review by DaveCTurner — APPROVED** [review]
LGTM with one suggestion

File: docs/reference/upgrade/rolling_upgrade.asciidoc (hunk 1 above)
Comment (pxsalehi) [reply docs/reference/upgrade/rolling_upgrade.asciidoc:20]: Yes. I wasn't entirely sure cross major versions we've had any changes to that endpoint (filtering by tier like that). So I left that out.

Comment (pxsalehi) [reply docs/reference/upgrade/rolling_upgrade.asciidoc:20]: somehow couldn't find it in the doc here: https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-nodes-info.html

Comment (DaveCTurner) [reply docs/reference/upgrade/rolling_upgrade.asciidoc]: This should work in 7.17 indeed. There's a link to the relevant page here:

> All the nodes selective options are explained [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster.html#cluster-nodes).

However, the linked page doesn't actually mention filtering by role. We should fix this too.

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/docs

**Pull request merged**